assets/js/app.js

```diff
@@ -97,6 +97,20 @@ $(".copy-button").on("click", function (event) {
   }
 })
 
+$(".click-to-reveal-content").on("click", ".reveal-button", function (event) {
+  let parent = $(this).parent();
+  let content = parent.data("content");
+  parent.text(content);
+  parent.append("<img src='/icons/hero-clipboard-document.svg' width='20px' class='inline-block copy-button cursor-pointer ms-2' />");
+  parent.append("<img src='/icons/hero-eye-slash.svg' width='20px' class='inline-block hide-button cursor-pointer ms-2' />");
+})
+
+$(".click-to-reveal-content").on("click", ".hide-button", function (event) {
+  let parent = $(this).parent();
+  parent.text("●●●●●●");
+  parent.append("<img src='/icons/hero-eye.svg' width='20px' class='inline-block reveal-button cursor-pointer ms-1' />");
+})
```

Comment:
nit: You may do JS part a little bit lighter.

The HTML part may have both revealed and hidden part. Just toggle them.
```html
<span class="click-to-reveal-content">
    <span class="hidden-content flex items-center">
        ●●●●●●
        <img src="/icons/hero-eye.svg" width="18px" class="inline-block reveal-button cursor-pointer ms1"/>
    </span>
    <span class="revealed-content flex items-center">
        SECRET CONTENT
        <img src="/icons/hero-eye-slash.svg" width="18px" class="inline-block hide-button cursor-pointer ms1"/>
    </span>
</span>
```

```javascript
$(".click-to-reveal-content").on("click", ".reveal-button", function (event) {
   $(this).parent().hide().siblings(".revealed-content").show();
});

$(".click-to-reveal-content").on("click", ".hide-button", function (event) {
   $(this).parent().hide().siblings(".hidden-content").show();
});

```